src/models/differentiable/evotree_model.jl

```diff
@@ -0,0 +1,147 @@
+    model::Any
+    likelihood::Symbol
+    function EvoTreeModel(model, likelihood)
+        if likelihood ∈ [:classification_binary, :classification_multi]
+            new(model, likelihood)
+        else
+            throw(
+                ArgumentError(
+                    "`type` should be in `[:classification_binary, :classification_multi]1.
+                    Support for regressors has not been implemented yet.`"
+                ),
+            )
+        end
+    end
+end
+
+"""
+Outer constructor method for EvoTreeModel.
+"""
+function EvoTreeModel(model::Any; likelihood::Symbol=:classification_binary)
+    return EvoTreeModel(model, likelihood)
+end
+
+# Methods
+"""
+    logits(M::MLJModel, X::AbstractArray)
+
+Calculates the logit scores output by the model M for the input data X.
+
+# Arguments
+- `M::MLJModel`: The model selected by the user. Must be a model from the MLJ library.
+- `X::AbstractArray`: The feature vector for which the logit scores are calculated.
+
+# Returns
+- `logits::Matrix`: A matrix of logits for each output class for each data point in X.
+
+# Example
+logits = Models.logits(M, x) # calculates the logit scores for each output class for the data point x
+"""
+function logits(M::EvoTreeModel, X::AbstractArray)
+    p = probs(M, X)
+    if M.likelihood == :classification_binary
+        logits = log.(p ./ (1 .- p))
+    else
+        logits = log.(p)
+    end
+    return logits
+end
+
+"""
+    probs(M::EvoTreeModel, X::AbstractArray{<:Number, 2})
+
+Calculates the probability scores for each output class for the two-dimensional input data matrix X.
+
+# Arguments
+- `M::EvoTreeModel`: The EvoTree model.
+- `X::AbstractArray`: The feature vector for which the predictions are made.
+
+# Returns
+- `p::Matrix`: A matrix of probability scores for each output class for each data point in X.
+
+# Example
+probabilities = Models.probs(M, X) # calculates the probability scores for each output class for each data point in X.
+"""
+function probs(M::EvoTreeModel, X::AbstractArray{<:Number,2})
+    output = MLJBase.predict(M.model, DataFrame(X', :auto))
+    p = pdf(output, classes(output))'
+    return p
+end
+
+"""
+    probs(M::EvoTreeModel, X::AbstractArray{<:Number, 1})
+
+Works the same way as the probs(M::EvoTreeModel, X::AbstractArray{<:Number, 2}) method above, but handles 1-dimensional rather than 2-dimensional input data.
+"""
+function probs(M::EvoTreeModel, X::AbstractArray{<:Number,1})
+    X = reshape(X, 1, length(X))
+    output = MLJBase.predict(M.model, DataFrame(X, :auto))
+    p = pdf(output, classes(output))'
+    return p
+end
+
+"""
+    probs(M::EvoTreeModel, X::AbstractArray{<:Number, 3})
+
+Works the same way as the probs(M::EvoTreeModel, X::AbstractArray{<:Number, 2}) method above, but handles 3-dimensional rather than 2-dimensional input data.
+"""
+function probs(M::EvoTreeModel, X::AbstractArray{<:Number,3})
+    output = SliceMap.slicemap(x -> probs(M, x), X; dims=[1, 2])
```

Comment:
I don't fully understand what's going on right here. I guess it could be useful to add a comment before that line and explain more or less what the `SliceMap.slicemap()` operation does and why the `dims` array is `[1, 2]`

Reply:
Yep, it's much cleaner now, thanks.